pkg/controller/command/sdscomm/sdscomm_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sdscomm
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/trustbloc/edv/pkg/restapi/messages"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/require"
+	"github.com/trustbloc/edv/pkg/edvprovider/memedvprovider"
+	"github.com/trustbloc/edv/pkg/restapi"
+)
+
+func TestSDSComm_CreateDIDVault(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		srvAddr := randomURL()
+
+		srv := startEDVServer(t, srvAddr)
+
+		waitForServerToStart(t, srvAddr)
+
+		sdsComm := New(fmt.Sprintf("http://%s/encrypted-data-vaults", srvAddr), "James")
+
+		err := sdsComm.CreateDIDVault()
+		require.NoError(t, err)
+
+		err = srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	})
+	t.Run("SDS server unreachable (unsupported protocol scheme provided)", func(t *testing.T) {
+		sdsComm := New("BadURL", "James")
+
+		err := sdsComm.CreateDIDVault()
+		require.Contains(t, err.Error(), "unsupported protocol scheme")
+	})
+}
+
+func TestSDSComm_StoreDIDDocument(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		srvAddr := randomURL()
+
+		srv := startEDVServer(t, srvAddr)
+
+		waitForServerToStart(t, srvAddr)
+
+		sdsComm := New(fmt.Sprintf("http://%s/encrypted-data-vaults", srvAddr), "James")
+
+		err := sdsComm.CreateDIDVault()
+		require.NoError(t, err)
+
+		sampleDIDDocData := DIDDocData{}
+
+		err = sdsComm.StoreDIDDocument(&sampleDIDDocData)
+		require.NoError(t, err)
+
+		err = srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	})
+	t.Run("Fail to store - vault not found", func(t *testing.T) {
+		srvAddr := randomURL()
+
+		srv := startEDVServer(t, srvAddr)
+
+		waitForServerToStart(t, srvAddr)
+
+		sdsComm := New(fmt.Sprintf("http://%s/encrypted-data-vaults", srvAddr), "James")
+
+		sampleDIDDocData := DIDDocData{}
+
+		err := sdsComm.StoreDIDDocument(&sampleDIDDocData)
+		require.NotNil(t, err)
+		require.Contains(t, err.Error(), messages.ErrVaultNotFound.Error())
+
+		err = srv.Shutdown(context.Background())
+		require.NoError(t, err)
+	})
+}
+
+func randomURL() string {
+	return fmt.Sprintf("localhost:%d", mustGetRandomPort(3))
+}
+
+func mustGetRandomPort(n int) int {
+	for ; n > 0; n-- {
+		port, err := getRandomPort()
+		if err != nil {
+			continue
+		}
+
+		return port
+	}
+	panic("cannot acquire the random port")
+}
+
+func getRandomPort() (int, error) {
```

Comment:
Just curious, couldn't you use httpmock server for unit tests?

Reply:
Please use httpmock for unit test

Reply:
Avoid starting server in unit test

Reply:
Done!